Mathlib/Data/Nat/Choose/Bounds.lean

```diff
@@ -50,4 +50,19 @@ theorem pow_le_choose (r n : ℕ) : ((n + 1 - r : ℕ) ^ r : α) / r ! ≤ n.cho
     exact n.pow_sub_le_descFactorial r
   exact mod_cast r.factorial_pos
 
+theorem choose_succ_le_two_pow {n k : ℕ} : (n + 1).choose k <= 2 ^ n := by
+  by_cases lt : n + 1 < k
+  · simp [choose_eq_zero_of_lt lt]
+  · cases' n with n
+    · cases k <;> simp_all
+    · cases' k with k
+      · exact Nat.one_le_two_pow
+      · calc
+          (n + 2).choose (k + 1) =
+            (n + 1).choose k +
+            (n + 1).choose (k + 1)           := by simp [choose_succ_succ']
+          _ ≤ 2 ^ n + (n + 1).choose (k + 1) := Nat.add_le_add_right choose_succ_le_two_pow _
+          _ ≤ 2 ^ n + 2 ^ n                  := Nat.add_le_add_left choose_succ_le_two_pow _
+          _ = 2 ^ (n + 1)                    := Eq.symm (two_pow_succ n)
```

Comment:
What you've got is fine and more readable, but just wanted to mention this branch could also be
```
      · rw [choose_succ_succ', two_pow_succ]
        exact Nat.add_le_add choose_succ_le_two_pow choose_succ_le_two_pow
```

Reply:
Personally I would use the shorter proof since all the newlines at the start of the calc block are a bit awkward.